### ISSUE TITLE
Update build.sh to build the same way as release manager will

### ISSUE THREAD

**Issue by Mpdreamz** [elastic/apm-server#7336]
Currently our [`Linux Build`](https://github.com/elastic/apm-server/blob/7c194b962f869fbae285f9750d795879839d1fa2/Jenkinsfile#L126) calls `.ci/scripts/build.sh` which in turn calls `make apm-server`. 

Should we update this so that `.ci/scripts/build.sh` invokes `sudo make release-manager-snapshot` or `sudo make release-manager` ? 

That way we are certain changes don't break the unified build.

**Comment by axw** [comment]
Given how infrequently we break release-manager, I don't think we should do this - at least not without a significant reduction in the amount of time taken to build release artifacts. PRs are already too slow to build.

**Comment by axw** [comment]
> All CI tasks should fail rarely though.

As mentioned above, we rarely break release-manager.

> Will it really add to PR build times?

The time taken for the "Package" stage in CI, which runs `make release` (dependency of `make release-manager-snapshot`), is 30-40 minutes: https://apm-ci.elastic.co/job/apm-server/job/apm-server-mbp/job/main/. I don't know what the difference is to your laptop (maybe CI is building more packages? maybe it's just a wimpy machine?), but that's far too long already.

We might be able to trim the time, but again we rarely break release-manager so this is not a useful exercise at the moment. Yes it does run in parallel, and It's a similar time to the "APM Integration Tests" stage, but I'm about to remove from the server CI.

When we move to DRA:
 - we will be building packages multiple times a day
 - we won't be blocking any other teams if they do fail
 - we will have notifications if they fail

The notifications should be enough for us to keep on top of any sporadic packaging failures. If any of the above changes, let's revisit this.

**Comment by Mpdreamz** [comment]
++ also just spotted https://github.com/elastic/apm-server/pull/7530 which makes this ticket no longer relevant.
